### PR TITLE
test(spin): add more integration tests

### LINKED ISSUE #127: spin shim needs more integration tests
The current integration tests for the spin shim is lacking. It only performs a GET request on the spin-hello-world image. We will want to add more integration tests for the spin shim. 
- [ ] Spin outbound redis test #181 
- [ ] Spin inbound redis (a.k.a redis trigger) test
- [ ] Spin OCI test
- [ ] Spin + Dapr with pod agonosity test

### PR DESCRIPTION
**Pull request by Mossaka**
This commit adds an integration test for spin-keyvalue using dynamic config file that was introduced by #179 and an integraion test for outbound redis introduced by #117

Closes #127

### CODE REVIEW

**Comment by Mossaka** [comment]
Could you please take a look? @jsturtevant

File: tests/workloads/spin-runtime-config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+data:
+  runtime-config.toml: "# This defines a new store named user_data\n[key_value_store.foo]\ntype
```
Comment (jsturtevant) [inline tests/workloads/spin-runtime-config.yaml:3]: nit: you can do 

```
config.yml: |-
  # This defines a new store named user_data
  [key_value_store.foo]
  type = "spin"  
  path = "user_data.db"

Comment (jsturtevant) [reply tests/workloads/spin-runtime-config.yaml:3]: Question: is this config checked anywhere in the tests to ensure it is used? I am not seeing it but might have missed it.

File: containerd-shim-spin/src/engine.rs [hunk 2]
```diff
@@ -223,7 +223,7 @@ impl SpinEngine {
         let mut runtime_config = RuntimeConfig::new(PathBuf::from("/").into());
         // Load in runtime config if one exists at expected location
         if Path::new(RUNTIME_CONFIG_PATH).exists() {
-            runtime_config.merge_config_file(RUNTIME_CONFIG_PATH);
```
Comment (jsturtevant) [inline containerd-shim-spin/src/engine.rs:226]: thought: We chatted about this yesterday, but we could make  `RUNTIME_CONFIG_PATH` an env (defaults to the root) and set an env in the yaml that points to a different location.  not blocking, just an option

File: tests/src/lib.rs [hunk 3]
```diff
@@ -8,15 +8,11 @@ use rand::{distributions::Alphanumeric, Rng};
 
 mod integration_test;
 
-pub async fn retry_get(
-    url: &str,
-    buf: &mut Vec<u8>,
-    retry_times: u32,
-    interval_in_secs: u64,
-) -> Result<()> {
+pub async fn retry_get(url: &str, retry_times: u32, interval_in_secs: u64) -> Result<Vec<u8>> {
```
Comment (jsturtevant) [inline tests/src/lib.rs:11]: nice improvement

File: containerd-shim-spin/src/engine.rs (hunk 2 above)
Comment (Mossaka) [reply containerd-shim-spin/src/engine.rs:226]: Yeah I remember this discussion. I can work on this in another PR. (nit: I prefer `SPIN_RUNTIME_CONFIG_PATH`)

File: tests/workloads/spin-runtime-config.yaml (hunk 1 above)
Comment (Mossaka) [reply tests/workloads/spin-runtime-config.yaml:3]: I believe `let store = Store::open("foo")?;` is using the `foo` store defined in the `runtime-config.toml`. The `images/spin-keyvalue/Dockerfile` does not copy runtime config file to the image, so it must have been reading the config from the ConfigMap mounted as a file.

Comment (Mossaka) [reply tests/workloads/spin-runtime-config.yaml:3]: Yeah this file is generated from `kubectl create configmap spin-runtime-config --from-file=runtime-config.toml --output=yaml`

**Comment by jsturtevant** [comment]
LGTM

**Pull request merged**